### DISCUSSION TITLE
Github pages wont deploy correctly only populates blank screen

### DISCUSSION THREAD

**Discussion by morganwestcoding** [community/community#52368] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi this is my first time trying to deploy a project to github pages, whenever I do it just shows a blank page. The project works when I do npm run dev but not npm run deploy, 
here is my main repo: https://github.com/morganwestcoding/morgan-west-portfolio
the github page appears to be deploying off of the branch gh page but for some reason all the files populated very weird in that branch. I am very new to this anything helps thanks.

**Comment by airtower-luna** [comment, marked as answer]
Take a look at the [`index.html` on your `gh-pages` branch](https://github.com/morganwestcoding/morgan-west-portfolio/blob/c2ebc2c3c819795b0b180a2434ef58991774a274/index.html). Its body is basically empty:

```html
<body> <div id="root"></div> <script src="/App.79369579.js"></script> </body>
```

I assume that the script is supposed to populate the document, but it can't be loaded because there's an incorrect absolute URL (starting with `/`). `App.79369579.js` exists in your repository, but it's deployed to `https://morganwestcoding.github.io/morgan-west-portfolio/App.79369579.js`, not `https://morganwestcoding.github.io/App.79369579.js` (where the `<script>` tag tells the browser to look). You'll need to either make the URL relative (so it works no matter under which path your site is deployed), or fix the base of the path.

**Reply by morganwestcoding** [reply]
This was a game changer, so now my page is populating my text but my images and fonts are not showing up. I tried also linking the css the way we connected the html but this did not work? Any suggestions?

**Reply by morganwestcoding** [reply]
If it helps at all when I was running dev my css linked was in my app.js

**Reply by airtower-luna** [reply]
If they're simply not loaded it sounds like there may be similar path issues. It's probably best if you check the documentation of the framework you use (React, as far as I can see at a glance) on how to either set the base directory for your site, or (preferably) have it generate relative paths.

**Reply by RyanCocking** [reply]
Making the paths relative finally caused my page to load, thank you! It's just a barebones Three.js project with some text. Previously, when visiting the page, pressing `F12` to bring up developer tools and checking the `Console` tab showed a 404 error, complaining that some built `.js` files could not be found. I was following a few tutorials online and somehow none of them mention that paths may cause issues.

As for how I implemented the fix, I created `vite.config.js` in my root directory:
```
import { defineConfig } from 'vite';

export default defineConfig({
base: './',              // ensure relative paths, e.g. in index.html
build: {
    outDir: 'dist',      // gets deployed to github pages
    assetsDir: 'assets', // static assets (images, js, css, etc.)
},
});
```

Of course `outDir` and `assetsDir` are set to their defaults, but at least I can be reminded of their existence!

**Comment by Fuzail22** [comment]
# Additional information:
- To get the right path automatically in your next build, add homepage attribute to your package.json file before your next build.
- below should be your homepage
   ```
   "homepage": "https://morganwestcoding.github.io/morgan-west-portfolio"
   ```
![sample package.json](https://github.com/community/community/assets/102959853/5290d59a-7c41-42e9-96f2-48a040854f2e)

**Reply by jyoung28** [reply]
Hey, I have this blank screen issue too, but the website initially works for a few minutes after deployment but then goes to a blank screen after a while. Do you have any idea what this could be? I already have the homepage fix in my package.json file, I changed any Browser Router to HashRouter. I am still stuck though. (React app using Rhouters)

**Reply by Fuzail22** [reply]
is this behaviours observed only after deployment? Since there are several possibilites, can you please share more information about this.  If possiible share a link to your repo.

**Reply by OscarCarCar** [reply]
I met this issue as well. What I do is to change the homepage path. In my case, I first do the following things 

1. The repository name is "username.github.io " 
2. I have add the "homepage" info in the package.jsong file 
3. I use gh-pages to deploy my web

I found that I cannot access the website, because the relative paths for accessing those js, css files are wrong, you can try to check it in the "index.html" file.

Then what I did is, justchange the "homepage" path, i.e. 
FROM    "homepage": "https://username.github.io/username.github.io"
TO          "homepage": "https://username.github.io/"

Finally, it works.

**Comment by prajaktakap00r** [comment]
I have done all the solutions that are mentioned above still its not working for me

**Reply by Leandro-pixel** [reply]
did you found the solution?

**Reply by prajaktakap00r** [reply]
yeah. Found a video on YouTube and followed the steps.

https://youtu.be/4G6O0BIoq6M?si=WgqiO914o5k3CYtc
https://youtu.be/7wzuievFjrk?si=xAepvLDpMkG7Z8IA

**Comment by gass-git** [comment]
I had a similar problem and took quite some time to figure it out, couldn't find much help online so I'll leave an explainer here in case it might help someone with the same problem.

**issue**: I was deploying to Github pages using a workflow action using `upload-pages-artifact@v3` and `deploy-pages@v4` and the site seemed to work fine for some time and then it would go blank. The problem would get fixed by re-running the workflow action but it would go blank again after some time.

During inspection I discovered that the site was fetching the `index.html` from` src/` and not from the **distribution folder**. Frustrated I asked myself how can this be !? if my action path is set to `dist/`

Well, the answer is that in the repository settings you should specify the source of the build and deployment. It can be from a branch or from an action. In my case I had it set from a branch, so that is why it we retrieving the wrong` index.html`

**In short, if you are having this issue check the source of build and deployment:**

Go to Settings > Pages in your GitHub repo. Ensure "Build and deployment > Source" is set to GitHub Actions and not "Deploy from a branch" if you are using a workflow.

**Comment by paiikarthik** [comment]
I also had Similar Problem. So i renamed  file as index.php and Deleted README.md then it worked For me.
